### DISCUSSION TITLE
List of apps with open connections

### DISCUSSION THREAD

**Discussion by random3456345** [evilsocket/opensnitch#1112] [Ideas]
Hello, 
something I would like to see is a list of apps that have active connections, similar to the command `sudo netstat -tup`, with the corresponding icons at the beginning of each line.

**Comment by gustavo-iniguez-goya** [comment]
Glad you asked for it, because I already had it in the roadmap, so I have an excuse to add it :)

We have two options: dump the active connections and/or dump the conntrack list (conntrack -L)

The daemon part is already done, but I still need to write the GUI part. I was in doubt about how to display the list, in a new tab, or in a new window (with buttons to kill connections, copy/paste, etc...).

tasks:
 - [x] (9e0f3a4797f8365b3616f663623ae182864e23ed) Write a TasksManager: an internal option to launch "tasks" from the GUI to the daemons:
    - [x] Monitor PIDs, sending back to the GUI information of a PID (already done).
    - [x] Monitor daemon/host details (https://pkg.go.dev/syscall#Sysinfo_t).
    - [x] Monitor opened connections (335f2a783d6d354840f9c4847a07172e51b711a0, c5e24c04f1f4d2ac03b7d6be5bc4ccc4f13854cb, ).
    - [ ] Allow to kill sockets.
    - [ ] Allow to create firewall rules from sockets.
    - [x] Display the path of the process.
    - [x] Allow to jump to the Hosts/IPs/Process tabs by clicking on the corresponding column of the netstat view (6e7aaf935b8f734962d727850736e6e5bca2f561).
    - ...
 - [x] List opened connections.
 - [ ]  #1143

**Reply by NRGLine4Sec** [reply]
I think a new tab next to "Applications" could be the right place for this.

**Reply by random3456345** [reply]
Glad my input was listened to. I wasn't expecting anything to come of it.

**Reply by gustavo-iniguez-goya** [reply]
@NRGLine4Sec , @random3456345 Feature added. Now you can list the connections of a system similar to ss or netstat. It's the first version, we'll keep improving it.

Some issues:
 - It seems that https://github.com/vishvananda/netlink/ lib has a bug listing some sockets, like those created with AF_PACKET.

Future improvements:
 - Allow to kill sockets.
 - Get the host of an IP (we should already have it in cache).
 - Display the path of the process (right now only the comm is displayed)
 - Allow to jump to the Hosts/IPs/Process tabs by clicking on the corresponding column of the netstat view.
 - Display an icon to indicate if a socket in LISTEN state has a firewall rule associated with it, in order to ease the task of creating rules for system services.

**Reply by random3456345** [reply]
> Feature added. 

Is it in version 1.6.6? I can't find it.

**Reply by gustavo-iniguez-goya** [reply]
@random3456345 in version 1.7.0, master branch only.
